.github/workflows/build-linux-packages.yml

```diff
@@ -0,0 +1,94 @@
+name: Build Linux Packages
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*.*.*
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+  schedule:
+    - cron: '0 4 * * *'
+
+permissions:
+  contents: read
+
+jobs:
+  pre_job:
+    name: Skip Duplicate Jobs Pre Job
+    runs-on: ubuntu-22.04
+    permissions:
+      actions: write  # Needed for skip-duplicate-jobs job
+      contents: read
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+      python_version: ${{ steps.define_constants.outputs.python_version }}
+      cache_version: ${{ steps.define_constants.outputs.cache_version }}
+      aws_region: ${{ steps.define_constants.outputs.aws_region }}
+      aws_private_key_name: ${{ steps.define_constants.outputs.aws_private_key_name }}
+      cicd_workflow: ${{ steps.define_constants.outputs.cicd_workflow }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@12aca0a884f6137d619d6a8a09fcc3406ced5281 # v4.0.0
+        with:
+          cancel_others: 'true'
+          github_token: ${{ github.token }}
+
+      - name: Checkout repository
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4
+
+      - id: define_constants
+        run: |
+          echo "python_version=3.8" >> "${GITHUB_OUTPUT}"
+          echo "cache_version=$GITHUB_REF_NAME" >> "${GITHUB_OUTPUT}"
+          echo "aws_region=us-east-1" >> "${GITHUB_OUTPUT}"
+          echo "cicd_workflow=${{ github.run_id }}-${{ github.run_number }}-${{ github.run_attempt }}" >> "${GITHUB_OUTPUT}"
```

Comment:
I think you only need one of github.run_id or github.run_number not both, ref: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs

Reply:
I agree. I was splitting old workflows and copied it the way it was to be sure nothing breaks. You never know what depends on this one. In this case it's probably ok to change it, but I prefer to keep it that way and possibly fix it later.